### PR TITLE
Feature/export cmake config package

### PR DESCRIPTION
**Pull request by ClausKlein**
### Added VERIFY_INTERFACE_HEADER_SETS too.
(and fixed compiler errors found with this check)

### Add a test to check if the exported config package is usable

may be tested with the `generic cmake workflow presets`:
```bash
bash-5.2$ cmake --workflow debug
bash-5.2$ cmake --workflow release
```

### CODE REVIEW

**Comment by ClausKlein** [comment]
see too https://github.com/beman-project/inplace_vector/issues/15

File: CMakeLists.txt [hunk 1]
```diff
@@ -3,46 +3,88 @@
...
+block()
+# copied from execution26:
+set(TARGET_NAME beman_inplace_vector)
+set(TARGET_NAMESPACE beman)
+# set(TARGET_PREFIX ${TARGET_NAMESPACE}.${TARGET_NAME}) # FIXME: not used yet? CK
+# set(TARGET_LIBRARY ${PROJECT_NAME})
+# set(TARGET_ALIAS ${TARGET_LIBRARY}::${TARGET_LIBRARY})
+set(TARGET_PACKAGE_NAME ${PROJECT_NAME}-config)
+set(TARGETS_EXPORT_NAME ${PROJECT_NAME}-targets)
+set(INSTALL_CONFIGDIR ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
```
Comment (ClausKlein) [inline CMakeLists.txt:51]: NOTE: this is not clear to me?

File: .cmake-format
Comment (wusatosi) [inline .cmake-format:1]: This project uses [gersemi](https://github.com/BlankSpruce/gersemi) with pre-commit for cmake formatting, it does not recognize this format file. 

CI needs to be updated to check the conformity of the project to its format configuration, thus there usually is sweeping change we need to apply to introduce a format definition, if this is something you really want to contribute, please open another PR specifically for introducing another formatting toolkit.

Otherwise, please remove this file from this PR.

File: CMakeLists.txt [hunk 2]
```diff
@@ -3,46 +3,88 @@
...
 
-cmake_minimum_required(VERSION 3.23)
+set(CMAKE_SKIP_TEST_ALL_DEPENDENCY FALSE)
 
-option(BUILD_TESTING "Build tests" ON)
+cmake_minimum_required(VERSION 3.25...3.31)
 
 project(
-    beman.inplace_vector
+    beman_inplace_vector
```
Comment (wusatosi) [inline CMakeLists.txt:11]: This does not conform to our naming tradition, why changing it to `beman_inplace_vector`?

We have used `beman.short_name` for almost all other projects.

[Exemplar](https://github.com/beman-project/exemplar/blob/849f0bdd39c189c6ca6e05544fb89d9b292d6b6e/CMakeLists.txt#L5-L10), [utf_view](https://github.com/beman-project/utf_view/blob/main/CMakeLists.txt#L9), [iterator_interface](https://github.com/beman-project/iterator_interface/blob/main/CMakeLists.txt#L8-L11),

File: include/beman/inplace_vector/inplace_vector.hpp [hunk 3]
```diff
@@ -540,7 +540,7 @@ class inplace_vector : public inplace_vector_base<T, Capacity> {
       for (; this->size() != Capacity && first != last; ++first) {
         emplace_back(*first);
       }
-      return frist;
+      return first;
```
Comment (wusatosi) [inline include/beman/inplace_vector/inplace_vector.hpp:543]: Nice catch

File: CMakeLists.txt [hunk 4]
```diff
@@ -3,46 +3,88 @@
...
+    DESCRIPTION "A dynamically-resizable vector with fixed capacity and embedded storage"
     LANGUAGES CXX
 )
 
-include(GNUInstallDirs)
+if(${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
+    message(FATAL_ERROR "In-source builds are not allowed!")
+endif()
+
+option(INPLACE_VECTOR_BUILD_TESTING "Build tests" ${PROJECT_IS_TOP_LEVEL})
```
Comment (wusatosi) [inline CMakeLists.txt:21]: Use `BEMAN_INPLACE_VECTOR_BUILD_TESTS` to conform with [[CMAKE.SKIP_TESTS]](https://github.com/beman-project/beman/blob/main/docs/BEMAN_STANDARD.md#cmake).

**Review by wusatosi — CHANGES_REQUESTED** [review]
Note the lint failure.

File: include/beman/inplace_vector/inplace_vector.hpp (hunk 3 above)
Comment (ClausKlein) [reply include/beman/inplace_vector/inplace_vector.hpp:543]: I use `codespell` ...

`pip install codespell`

File: CMakeLists.txt (hunk 2 above)
Comment (ClausKlein) [reply CMakeLists.txt:11]: I changed it because I needed  it for `cmake export config package`

And it was used like this in `excecuter26` project.

File: CMakeLists.txt (hunk 4 above)
Comment (ClausKlein) [reply CMakeLists.txt:21]: OK

File: .cmake-format
Comment (ClausKlein) [reply .cmake-format:1]: which file should i delet?
I do not format files by hand!
How can i localy fix the format?
I use clang-format to do this.

File: CMakeLists.txt (hunk 2 above)
Comment (ClausKlein) [reply CMakeLists.txt:11]: That will end in a very strange export package, not very common:
```bash
bash-5.2$ tree /usr/local/lib/cmake/boost_
boost_atomic-1.86.0/                    boost_graph_parallel-1.86.0/            boost_math_tr1l-1.86.0/                 boost_system-1.86.0/
boost_charconv-1.86.0/                  boost_headers-1.86.0/                   boost_nowide-1.86.0/                    boost_test_exec_monitor-1.86.0/
boost_chrono-1.86.0/                    boost_iostreams-1.86.0/                 boost_prg_exec_monitor-1.86.0/          boost_thread-1.86.0/
boost_container-1.86.0/                 boost_json-1.86.0/                      boost_process-1.86.0/                   boost_timer-1.86.0/
boost_context-1.86.0/                   boost_locale-1.86.0/                    boost_program_options-1.86.0/           boost_type_erasure-1.86.0/
boost_contract-1.86.0/                  boost_log-1.86.0/                       boost_random-1.86.0/                    boost_unit_test_framework-1.86.0/
boost_coroutine-1.86.0/                 boost_log_setup-1.86.0/                 boost_regex-1.86.0/                     boost_url-1.86.0/
boost_date_time-1.86.0/                 boost_math_c99-1.86.0/                  boost_serialization-1.86.0/             boost_wave-1.86.0/
boost_exception-1.86.0/                 boost_math_c99f-1.86.0/                 boost_stacktrace_addr2line-1.86.0/      boost_wserialization-1.86.0/
boost_fiber-1.86.0/                     boost_math_c99l-1.86.0/                 boost_stacktrace_basic-1.86.0/          
boost_filesystem-1.86.0/                boost_math_tr1-1.86.0/                  boost_stacktrace_from_exception-1.86.0/ 
boost_graph-1.86.0/                     boost_math_tr1f-1.86.0/                 boost_stacktrace_noop-1.86.0/           
bash-5.2$ tree /usr/local/lib/cmake/boost_atomic-1.86.0
/usr/local/lib/cmake/boost_atomic-1.86.0
|-- boost_atomic-config-version.cmake
|-- boost_atomic-config.cmake
|-- libboost_atomic-variant-mt-shared.cmake
`-- libboost_atomic-variant-mt-static.cmake

1 directory, 4 files
bash-5.2$ tree stagedir/lib/cmake/
stagedir/lib/cmake/
`-- beman_inplace_vector
    |-- beman_inplace_vector-config-version.cmake
    |-- beman_inplace_vector-config.cmake
    `-- beman_inplace_vector-targets.cmake

2 directories, 3 files
bash-5.2$ 
```

Comment (ClausKlein) [reply CMakeLists.txt:11]: and it may be used like this:
```cmake
#---------------------------------------------------------------------------------------
# search required packages and libs
#---------------------------------------------------------------------------------------

find_package(
    Boost 1.86 CONFIG
    COMPONENTS filesystem asio headers
    REQUIRED
)
```

File: .cmake-format
Comment (wusatosi) [reply .cmake-format:1]: you can just run `pre-commit run --all-files`, its automatic.

I don't think clang-format formats cmake files?

You can just delete `.cmake-format`

**Review by wusatosi — COMMENTED** [review]
Thank you for your contribution.

I am not qualified to review CMake files so I cannot single-handly merge/ approve this.

Also, can we defer presets and CI updates to another PR?

I would suggest you limit the scope of your contribution in this PR to be:
- `CMakeLists.txt`

You should split other changes into two PR sets:
1. CI Improvement:
    - macos + windows CI test
2. Introduce presets (I will hold off to this before [we figure out a standard preset set](https://discourse.bemanproject.org/t/what-do-we-want-our-presets-to-be/246))

Just a note:
Currently we have a infrastructure syncing issue as we have a lot of duplicate work going on on shared infrastructure like CI config, CMake files and CMake presets.

We are going to figure out a way to have a central source of truth/ reference on infrastructure and all the projects will sync infrastructure code off that.

My bet is that will either be in a standalone repository or in `exemplar`, I will advise you to contribute to that repo on infrastructure related improvements. I believe it have a higher priority on review as well.

If your improvements are accepted in `exemplar`, it would be easier for me who has less experience to reference that PR and give you a green light.

I don't want to incur more duplicated work so I would hold-off contributing here before we figure out this infrastructure syncing issue.

Again I appreciate your contribution, you seem to know a lot of CMake and its best practices. I would advise you to come join our weekly sync meeting to get your improvement suggestions discussed and accepted quicker.

File: .cmake-format
Comment (ClausKlein) [reply .cmake-format:1]: OK, are you sure? I will have a try.

File: CMakeLists.txt (hunk 2 above)
Comment (DeveloperPaul123) [reply CMakeLists.txt:11]: I would also agree that using `_` is more idiomatic than `.` within a project/target name.

**Comment by ClausKlein** [comment]
OK

**Comment by ClausKlein** [comment]
> You should split other changes into two PR sets:

CI Improvement:
- macos + windows CI test
  
  In my experience splitting **CI** workflows files in **OS** specific parts makes some task like install toolchain really simpler!

- Introduce presets 

Here and in execution26 the I found no preset.
So I created the simples preset possible: a `generic` one with only tow **workflows** `debug` and `release` which works on each **OS** and with each toolchain. Just as demo and as prove of concept.

The preset used in `examplar` will be more and more complex and unreadable!
It needs a `workflow` for each compiler and each build config type that should be supported.

> Just a note:
> Currently we have an infrastructure syncing issue as we have a lot of duplicate work going on on shared infrastructure like > CI config, CMake files and CMake presets.

**Comment by ClausKlein** [comment]
Updated prove of concept for `cmake workflow presets`:
```bash
bash-5.2$ cmake --workflow gcov
Executing workflow step 1 of 2: configure preset "gcov"

Preset CMake variables:

  CMAKE_BUILD_TYPE="Coverage"
  CMAKE_CONFIGURATION_TYPES="Debug;RelWithDebInfo;Coverage;Asan;Lsan;Usan"
  CMAKE_CXX_COMPILER="/usr/local/opt/llvm/bin/clang++"
  CMAKE_CXX_EXTENSIONS:BOOL="FALSE"
  CMAKE_CXX_FLAGS="-Wall -Wextra -Wpedantic"
  CMAKE_CXX_FLAGS_COVERAGE="-g --coverage"
  CMAKE_CXX_STANDARD="23"
  CMAKE_CXX_STANDARD_REQUIRED:BOOL="TRUE"
  CMAKE_C_COMPILER="/usr/local/opt/llvm/bin/clang"
  CMAKE_C_FLAGS="-Wall -Wextra -Wpedantic"
  CMAKE_C_FLAGS_COVERAGE="-g --coverage"
  CMAKE_EXPORT_COMPILE_COMMANDS:BOOL="TRUE"
  CMAKE_INSTALL_PREFIX:PATH="/Users/clausklein/Workspace/cpp/beman-project/inplace_vector/stagedir"
  CMAKE_LD_FLAGS_COVERAGE="--coverage"
  CMAKE_PREFIX_PATH:STRING="/Users/clausklein/Workspace/cpp/beman-project/inplace_vector/stagedir"
  CMAKE_SKIP_TEST_ALL_DEPENDENCY:BOOL="FALSE"
  CMAKE_TOOLCHAIN_FILE:STRING="/Users/clausklein/Workspace/cpp/beman-project/inplace_vector/cmake/toolchains/default.cmake"
  GCOV_EXECUTABLE="llvm-cov gcov"

-- Examples to be built: fibonacci
-- Configuring done (0.3s)
-- Generating done (0.1s)
-- Build files have been written to: /Users/clausklein/Workspace/cpp/beman-project/inplace_vector/build

Executing workflow step 2 of 2: build preset "gcov"

[2/4] Running tests...
Test project /Users/clausklein/Workspace/cpp/beman-project/inplace_vector/build
    Start 1: beman.inplace_vector.test
1/2 Test #1: beman.inplace_vector.test ........   Passed    0.15 sec
    Start 2: fibonacci
2/2 Test #2: fibonacci ........................   Passed    0.05 sec

100% tests passed, 0 tests failed out of 2

Total Test time (real) =   0.21 sec
[4/4] Running gcovr to process coverage results
(INFO) - MainThread - Reading coverage data...
(INFO) - MainThread - Writing coverage report...
lines: 92.6% (100 out of 108)
functions: 100.0% (37 out of 37)
branches: 42.6% (23 out of 54)
bash-5.2$ cmake --workflow --list-presets
Available workflow presets:

  "asan"
  "lsan"
  "usan"
  "gcov"
  "debug"
  "release"
```
### same `generic presets` on every build host available using `CXX` environment variable:
```
bash-5.2$ cmake --list-presets
Available configure presets:

  "debug"   - Debug Build
  "release" - Release Build
  "gcov"    - Coverage Build
  "asan"    - AddressSanitizer Build
  "lsan"    - LeakSanitizer Build
  "usan"    - UndefinedSanitizer Build
bash-5.2$ 
```

File: CMakeLists.txt [hunk 5]
```diff
@@ -27,41 +33,102 @@ option(
...
-    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${CMAKE_PROJECT_NAME}
-    FILES_MATCHING
-    PATTERN
-    "${CMAKE_CURRENT_SOURCE_DIR}/include/beman/inplace_vector/inplace_vector.hpp"
-)
+    set(TARGET_PACKAGE_NAME ${PROJECT_NAME}-config)
+    set(TARGETS_EXPORT_NAME ${PROJECT_NAME}-targets)
+    set(INSTALL_CONFIGDIR ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
+
+    # Install the InplaceVector library to the appropriate destination
```
Comment (wusatosi) [inline CMakeLists.txt:78]: Need help reviewing this

File: .github/workflows/ci_tests.yml
Comment (wusatosi) [inline .github/workflows/ci_tests.yml:1]: Change reject out of scope, please move to another PR.

**Comment by wusatosi** [comment]
> [ OS per workflow ]

I honestly have no problem with both approaches, the reason I am trying to solve all configuration into one file in exemplar is so it is clear what platform gets what kind of compile flags exceptions vs what testing matrix is universal in one file. And it's also easier for downstream to copy paste that single file to create a new matrix (so there could be multiple domains of CI testing files). I don't want all my testing configs scattered around multiple files.

> [ Preset design ]

I can definitely see where you are coming from. And I do agree with you to an extent. Please leave this as a comment on the exemplar repo or the discourse thread.

File: .github/workflows/macos.yml
Comment (wusatosi) [inline .github/workflows/macos.yml:1]: Change rejected due to out of scope, please move to another PR.

File: .github/workflows/windows.yml
Comment (wusatosi) [inline .github/workflows/windows.yml:1]: Change rejected due to out of scope, please move to another PR.

File: CMakePresets.json
Comment (wusatosi) [inline CMakePresets.json:1]: Change rejected due to out of scope, please move to another PR.

File: cmake/CMakeDarwinPresets.json
Comment (wusatosi) [inline cmake/CMakeDarwinPresets.json:1]: Change rejected due to out of scope, please move to another PR.

File: cmake/CMakeGenericPresets.json
Comment (wusatosi) [inline cmake/CMakeGenericPresets.json:1]: Change rejected due to out of scope, please move to another PR.

File: cmake/CMakeLinuxPresets.json
Comment (wusatosi) [inline cmake/CMakeLinuxPresets.json:1]: Change rejected due to out of scope, please move to another PR.

File: cmake/CMakeWindowsPresets.json
Comment (wusatosi) [inline cmake/CMakeWindowsPresets.json:1]: Change rejected due to out of scope, please move to another PR.

File: cmake/gcovr.cfg.in
Comment (wusatosi) [inline cmake/gcovr.cfg.in:1]: Change rejected due to out of scope, please move to another PR.

File: cmake/toolchains/default.cmake
Comment (wusatosi) [inline cmake/toolchains/default.cmake:1]: Change rejected due to out of scope, please move to another PR.

**Review by wusatosi — CHANGES_REQUESTED** [review]
Hey since I kept seeing updates from this PR I decided to jump in and check again. 

The CMake related change looks good, again I cannot single handedly review them but I've made my best effort to go through them and to make them obvious to appropriate reviewers.

I again advise you to move the following changes out of this PR:

- CI changes
- Preset addition
- coverage info generation 

Again those improvements are badly needed. But please keep them separate into smaller PR so we can review your changes quicker. e. g. If you submit a PR that only introduce CI/ code coverage, I probably could have it merged without request help from others in beman. 

I suggest you keep only the following files in this PR:
- main CMake file
- examples/ and include/ makefile
- CMake file needed to generate package.

I value your effort and your time so I do really want to have your changes merged. So I again request you to break the PR down to smaller constituents, so we can have them reviewed ASAP.

I suggest you open the code coverage related change in another PR first, as that seems to be the smallest impacting change. I would be comfortable reviewing the change and sync your improvements upstream to exemplar. 

Our next weekly sync happens on Monday 12PM EST, I sincerely invite you to come to join us. I think your suggestion for preset is a good alternative to what we are trying to do right now. You seem to have great knowledge about CMake, it would be fantastic to talk through what you would suggest generally in the meeting and gain a consensus with us on what would be desired. The information on how to join us should be available on discourse, you can message @ leads on discourse if you have questions subscribing to the weekly sync calendar, I believe you should be able to join the meeting with only a link.

File: CMakeLists.txt [hunk 6]
```diff
@@ -27,41 +33,102 @@ option(
...
-    include(CTest)
+    enable_testing()
     add_subdirectory(tests/beman/inplace_vector)
 endif()
 
 if(BEMAN_EXEMPLAR_BUILD_EXAMPLES)
     add_subdirectory(examples)
 endif()
+
+# Coverage
```
[the remark this answers is not in the document]
Comment (ClausKlein) [reply CMakeLists.txt:124]: yes

File: tests/beman/inplace_vector/CMakeLists.txt [hunk 7]
```diff
@@ -6,6 +6,27 @@
...
+            ${CMAKE_CTEST_COMMAND} --verbose --output-on-failure -C Debug
+            --build-and-test "${PROJECT_SOURCE_DIR}/examples"
+            "${CMAKE_CURRENT_BINARY_DIR}/find-package-test" --build-generator
+            ${CMAKE_GENERATOR} --build-makeprogram ${CMAKE_MAKE_PROGRAM}
+            --build-options "-DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}"
+            "-DCMAKE_CXX_STANDARD=${CMAKE_CXX_STANDARD}"
+            "-DCMAKE_BUILD_TYPE=Debug"
+            "-DCMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH}"
+    )
+endif()
```
Comment (ClausKlein) [inline tests/beman/inplace_vector/CMakeLists.txt:32]: This add a test that check, if the our `examples standalone project` can be build with our  exported `cmake config package`.

File: tests/beman/inplace_vector/CMakeLists.txt [hunk 8]
```diff
@@ -6,6 +6,27 @@
...
+    beman.inplace_vector.test
+    PRIVATE beman::beman_inplace_vector
+)
 
 add_test(NAME beman.inplace_vector.test COMMAND beman.inplace_vector.test)
+
+# test if the targets are findable from the build directory
+# NOTE: For an INTERFACE library and multi config generators, we may always use
+# the -C Debug config type instead of  $<CONFIG> to prevent problems! CK
+if(CMAKE_BUILD_TYPE STREQUAL Debug)
```
[the remark this answers is not in the document]
Comment (ClausKlein) [reply tests/beman/inplace_vector/CMakeLists.txt:19]: At the moment, the `find-package-test` generates only the default **CMAKE_CONFIGURATION_TYPES**.
For header only libs, there is no need to check each config type with `$<CONFIG>`

File: tests/beman/inplace_vector/CMakeLists.txt [hunk 9]
```diff
@@ -6,6 +6,27 @@
 # Tests
 add_executable(beman.inplace_vector.test inplace_vector.test.cpp)
 
-target_link_libraries(beman.inplace_vector.test PRIVATE beman.inplace_vector)
+target_link_libraries(
+    beman.inplace_vector.test
+    PRIVATE beman::beman_inplace_vector
```
[the remark this answers is not in the document]
Comment (ClausKlein) [reply tests/beman/inplace_vector/CMakeLists.txt:11]: No, it is better to check the ALIAS to be consistent.

File: examples/CMakeLists.txt [hunk 10]
```diff
@@ -12,6 +21,7 @@ foreach(example ${ALL_EXAMPLES})
     )
     target_link_libraries(
         beman.inplace_vector.examples.${example}
-        beman.inplace_vector
+        beman::beman_inplace_vector
     )
+    add_test(NAME ${example} COMMAND beman.inplace_vector.examples.${example})
```
[the remark this answers is not in the document]
Comment (ClausKlein) [reply examples/CMakeLists.txt:26]: This in crease the `code coverage` and makes sure, the all `needed runtimes dependencies `are available.

File: examples/CMakeLists.txt [hunk 11]
```diff
@@ -1,8 +1,17 @@
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
+cmake_minimum_required(VERSION 3.25...3.31)
+
+project(beman_inplace_vector_example LANGUAGES CXX)
```
[the remark this answers is not in the document]
Comment (ClausKlein) [reply examples/CMakeLists.txt:5]: see tests ...

File: cmake/gcovr.cfg.in
Comment (ClausKlein) [reply cmake/gcovr.cfg.in:1]: I do a separate PR for cmake export config package ...

File: CMakePresets.json
Comment (ClausKlein) [reply CMakePresets.json:1]: yes

File: cmake/CMakeDarwinPresets.json
Comment (ClausKlein) [reply cmake/CMakeDarwinPresets.json:1]: yes

File: cmake/CMakeGenericPresets.json
Comment (ClausKlein) [reply cmake/CMakeGenericPresets.json:1]: yes

File: CMakeLists.txt [hunk 12]
```diff
@@ -27,41 +33,102 @@ option(
...
-    TARGETS beman.inplace_vector
-    EXPORT ${TARGETS_EXPORT_NAME}
-    DESTINATION
-    ${CMAKE_INSTALL_LIBDIR}
-)
+# export cmake config package
+block()
+    # NOTE: copied from execution26:
+    # FIXME: but not yet used? CK
+    # set(TARGET_NAME inplace_vector)
```
[the remark this answers is not in the document]
Comment (ClausKlein) [reply CMakeLists.txt:68]: ### That a long odyssey: 
This block was copied from execution26. But the original version use [CMAKE_PROJECT_NAME](https://cmake.org/cmake/help/latest/variable/CMAKE_PROJECT_NAME.html) and that is wrong!

If you import this project with `FETCH_CONTENT` the [PROJECT_NAME](https://cmake.org/cmake/help/latest/variable/PROJECT_NAME.html#variable:PROJECT_NAME) must be used.

File: CMakeLists.txt (hunk 5 above)
Comment (ClausKlein) [reply CMakeLists.txt:78]: Yes, most people do this **wrong** or only **partly** ....

File: CMakeLists.txt (hunk 2 above)
Comment (ClausKlein) [reply CMakeLists.txt:11]: ### The most important point in this **PR**!

So you agree this change?

File: CMakeLists.txt [hunk 13]
```diff
@@ -27,41 +33,102 @@ option(
...
 
+# includes
 include(GNUInstallDirs)
+include(CMakePackageConfigHelpers)
 
-add_library(beman.inplace_vector INTERFACE)
+add_library(beman_inplace_vector INTERFACE)
 # [CMAKE.LIBRARY_ALIAS]
-add_library(beman::inplace_vector ALIAS beman.inplace_vector)
+add_library(beman::beman_inplace_vector ALIAS beman_inplace_vector)
```
[the remark this answers is not in the document]
Comment (ClausKlein) [reply CMakeLists.txt:42]: see https://github.com/bemanproject/inplace_vector/pull/14#discussion_r1855406400

File: cmake/gcovr.cfg.in
Comment (ClausKlein) [reply cmake/gcovr.cfg.in:1]: yes

File: CMakeLists.txt (hunk 6 above)
[the remark this answers is not in the document]
Comment (ClausKlein) [reply CMakeLists.txt:124]: done

**Comment by ClausKlein** [comment]
to be continuers with other PR ...

File: CMakeLists.txt (hunk 2 above)
Comment (wusatosi) [reply CMakeLists.txt:11]: Hey, according to: https://github.com/bemanproject/beman/pull/69

We cannot update the project/ target name.
